File: app/src/main/java/c/m/storyapp/login/data/remote/dto/LoginResponseDTO.kt
package c.m.storyapp.login.data.remote.dto

import com.google.gson.annotations.SerializedName

data class LoginResponseDTO(
    @SerializedName("error")
    val error:Boolean,
    @SerializedName("message")
    val message: String,
    @SerializedName("loginResult")
    val loginResult: LoginResultDTO,
)

File: lib/login/data/remote/dto/login_response_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:story_app/login/data/remote/dto/login_result_dto.dart';

part 'login_response_dto.freezed.dart';
part 'login_response_dto.g.dart';

@freezed
class LoginResponseDTO with _$LoginResponseDTO {
  const factory LoginResponseDTO({
    @JsonKey(name: 'error') required bool error,
    @JsonKey(name: 'message') required String message,
    @JsonKey(name: 'loginResult') LoginResultDTO? loginResultDTO,
  }) = _LoginResponseDTO;

  factory LoginResponseDTO.fromJson(Map<String, Object?> json) =>
      _$LoginResponseDTOFromJson(json);
}
